homeassistant/components/zwave_js/config_flow.py

```diff
@@ -288,16 +258,95 @@ async def async_step_start_addon(
         )
 
         return self.async_show_form(
-            step_id="start_addon", data_schema=data_schema, errors=errors
+            step_id="configure_addon", data_schema=data_schema, errors=errors
         )
 
+    async def async_step_start_addon(
+        self, user_input: Optional[Dict[str, Any]] = None
+    ) -> Dict[str, Any]:
+        """Start Z-Wave JS add-on."""
+        assert self.hass
+        if not self.start_task:
+            self.start_task = self.hass.async_create_task(self._async_start_addon())
+            return self.async_show_progress(
+                step_id="start_addon", progress_action="start_addon"
+            )
+
+        try:
+            await self.start_task
+        except (CannotConnect, HassioAPIError) as err:
+            _LOGGER.error("Failed to start Z-Wave JS add-on: %s", err)
+            return self.async_show_progress_done(next_step_id="start_failed")
+
+        return self.async_show_progress_done(next_step_id="finish_addon_setup")
+
+    async def async_step_start_failed(
+        self, user_input: Optional[Dict[str, Any]] = None
+    ) -> Dict[str, Any]:
+        """Add-on start failed."""
+        return self.async_abort(reason="addon_start_failed")
+
+    async def _async_start_addon(self) -> None:
+        """Start the Z-Wave JS add-on."""
+        assert self.hass
+        try:
+            await async_start_addon(self.hass, ADDON_SLUG)
+            # Sleep some seconds to let the add-on start properly before connecting.
+            for _ in range(ADDON_SETUP_TIMEOUT_ROUNDS):
+                await asyncio.sleep(ADDON_SETUP_TIMEOUT)
```

Comment:
This looks a little brittle but its existing

Reply:
We don't currently have a way for add-ons to signal that they started successful. We can start the container, but that doesn't mean it starts correctly, ie if the wrong stick is configured.